### ISSUE TITLE
Questions about the data / models

### ISSUE THREAD

**Issue by shmsw25** [ruiqi-zhong/Meta-tuning#1]
Hi @ruiqi-zhong, thank you for such an awesome work and for releasing the data and the code!

I am very interested in using your setup, and have a few questions.

1. Do you have the corresponding original data, before being converted to a QA format? (so, just a classification format) I know you used the public data but I'm curious if there's a 1-1 mapping between the original datapoint and the converted datapoint.
2. You've mentioned that test datasets were subsampled. Are test datasets in the released data already subsampled, or are they the original ones before being subsampled? If it is the latter, do you plan to release the subsampled one?
3. I have a question on what the model does at inference time. Consider a binary classification task with an input sentence "This movie is amazing." Based on my understanding, this datapoint can be converted to two question-answer pairs:
* {q1="This movie is amazing. Positive Review?", a1="Yes"}
* {q2="This movie is amazing. Negative Review?", a2="No"}
My understanding is you randomly & uniformly sample between two pairs during meta-tuning. What happens at inference time (both without and with meta-tuning)? I guess you can compute four values: P(Yes|q1), P(No,q1), P(Yes|q2), P(No|q2). How do you use these four values to make a decision? (Apologies if this is already clarified in the paper and I've missed it.)
(Plus, more clarification for the format of the test data would be appreciated - I was not able to interpret it and how they are fed into the model.)

Thanks!

**Comment by ruiqi-zhong** [comment]
> Hi @ruiqi-zhong, thank you for such an awesome work and for releasing the data and the code!
> 
> I am very interested in using your setup, and have a few questions.
> 
> 1. Do you have the corresponding original data, before being converted to a QA format? (so, just a classification format) I know you used the public data but I'm curious if there's a 1-1 mapping between the original datapoint and the converted datapoint.

There might not be a 1-1 mapping between data in testing_dicts and the original public dataset for many of the datasets, but it is most likely that there is a correspondence between my own classification-format-data (not released, before conversion) and the public data. However, since we are human-resource constrained, we did not put a lot of energy on ensuring 1-1 mapping between the original dataset and our data. So you might need to implement the conversion yourself if you want a more reliable comparison. Please let me know if the format for the test data is confusing and I'd happy to explain more. 

> 2. You've mentioned that test datasets were subsampled. Are test datasets in the released data already subsampled, or are they the original ones before being subsampled? If it is the latter, do you plan to release the subsampled one?

The ones I released are the subsampled ones (i.e. the ones I used in my paper). 

> 3. I have a question on what the model does at inference time. Consider a binary classification task with an input sentence "This movie is amazing." Based on my understanding, this datapoint can be converted to two question-answer pairs:
> 
> * {q1="This movie is amazing. Positive Review?", a1="Yes"}
> * {q2="This movie is amazing. Negative Review?", a2="No"}
>   My understanding is you randomly & uniformly sample between two pairs during meta-tuning. What happens at inference time (both without and with meta-tuning)? I guess you can compute four values: P(Yes|q1), P(No,q1), P(Yes|q2), P(No|q2). How do you use these four values to make a decision? (Apologies if this is already clarified in the paper and I've missed it.)

In fact, there might be more than 4 values - since for each label there might be multiple questions/prompts. In Section 5.1, we discarded all the logits associated with "no", and for each question we evaluate the AUC-ROC separately by treating "Yes" as the positive class. In Section 5.3,  for each label, we first ensemble the predictions using different label descriptions, then the label with the highest "Yes" probability is selected. Therefore, given the four values you mentioned we will obtain 2 AUC-ROC scores.

This means prompt selection can still make a difference. For example, in a separate experiment that I didn't mention in our paper, on SST-2 dev set, we tried 4 different ways of asking whether the review is positive, and 4 different ways of asking whether the review is negative. Meta-tuned RoBERTa-Large gets 88-90% for positive ways of asking, and gets 82-86% for negative ways of asking. 

>   (Plus, more clarification for the format of the test data would be appreciated - I was not able to interpret it and how they are fed into the model.)

I just added some quick comments (see commit history to see changes). Feel free to let me know if anything is unclear and I will add more. Thanks!
> 
> Thanks!

**Comment by shmsw25** [comment]
Thank you for the detailed responses! I will take a closer look and will let you know if I have further questions.
